layers/+lang/lua/funcs.el

```diff
@@ -24,7 +24,13 @@
     "sr" 'lua-send-region
     "'" 'lua-show-process-buffer)
   (pcase lua-backend
-    (`lsp-emmy (spacemacs//lua-setup-lsp-emmy))))
+    ('lsp-emmy-lua (spacemacs//lua-setup-lsp-emmy-lua))
+    ('lsp-lua-language-server (spacemacs//lua-setup-lsp-lua-language-server))
+    ('lsp-lua-lsp (spacemacs//lua-setup-lsp-lua-lsp))
+    (_ (if lua-backend
+           (user-error "Unexpected value of `lua-backend', %s" lua-backend)
+         (when (configuration-layer/layer-used-p 'lsp)
```

Comment:
The last `when` clause results in undefined behaviour that which of the lua LSP client will be used is not clear.

```elisp
(setq-local lsp-disabled-clients
            '(emmy-lua lsp-lua-language-server lsp-lua-lsp))
(if (member lua-backend lsp-disabled-clients)
    (delq lua-backend lsp-disabled-clients)
  (user-error "Unexpected value of `lua-backend', %s" lua-backend))
```

Reply:
@lebensterben The last `when` clause will fallback to default behavior for user who just enable `lsp` and `lua` layer without any more configurations, `lsp` will try detect the `emmy-lua`, `lsp-lua-language-server`, `lsp-lua-lsp` in PATH or not.

Reply:
That's exactly what undefined behavior is.

Reply:
Understood your points, and I also understood `lsp-mode` design that less configuration, more automation.

Reply:
I'll add more documents in `Readme.org` to make it exactly how `lsp-mode` work in `lua` layer.

Reply:
Disagree.
If something is not working as expected, users should be clearly notified.

It's possible the user has multiple clients installed and he has a typo in dotspacemacs. Then your current commit would rely on lsp-mode to figure out which client to use, and that may not be the one user wanted.

Since there's no message about the invalid config, the use will be confused why lsp-mode is not using the clien@t he specified.

Reply:
When user give wrong configuration, current code will give `user-error` message. 
When user just enable `lsp` and `lua` layer without other configuration, what should Spacemacs do? Spacsmacs should have default behavior. 
The default behavior can do warn with a message, or, try its best to find a working set.

Reply:
Many Emacs user begin their Emacs experience from Spacemacs than other emacs framework (eg: doom-emacs) for Spacemacs is simple and elegant, especially it's usable with the default configuration.
Exactly configurations are necessary for experts; while default configurations keep friendly for freshman is worth also.

Reply:
But here's a possible scenario why the "default behaviour" actually results in unexpected behaviours.

1. Suppose multiple lua clients, `a` and `b`, are installed under two different locations, `A` and `B`.
2. Suppose, for example, uses `bash`.
   - `.bash_profile` is loaded for login shell only, and it has `export PATH=A`.
   - `.bashrc` is loaded for interactive shell only, and it has `export PATH=$PATH:B`.
3. Suppose user uses a graphical login manager, and environment variables sets in
   `.bash_profile` is not inherited by any child process of the login manager.
   - When user opens a terminal, it runs a non-login, interactive `bash` shell. And `PATH=B`, hence `b` is found first and used as the LSP client.
   - When user connects to this host via `ssh`, it runs a login, interactive shell. So `PATH=A:B`, and then `a` is found first and used as the LSP client.

In this simple scenario, `lsp-mode` would use different LSP client, without showing any explicit message.

Since each LSP client may provide different [capabilities](https://docs.microsoft.com/en-us/visualstudio/extensibility/language-server-protocol?view=vs-2019#capabilities), some commands in Emacs may work differently. Then it could be a huge pain for the user to figure out why sometimes a command doesn't work as expected.